src/Microsoft.Framework.ConfigurationModel.Json/JsonConfigurationExtension.cs

```diff
@@ -1,13 +1,33 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.Framework.ConfigurationModel.Json;
+using System;
+using System.IO;
+
 namespace Microsoft.Framework.ConfigurationModel
 {
     public static class JsonConfigurationExtension
     {
         public static IConfigurationSourceRoot AddJsonFile(this IConfigurationSourceRoot configuration, string path)
         {
-            configuration.Add(new JsonConfigurationSource(path));
+            return AddJsonFile(configuration, path, false);
+        }
+
+        public static IConfigurationSourceRoot AddJsonFile(this IConfigurationSourceRoot configuration, string path, bool ignoreMissingFile)
```

Comment:
Can we change ignoreMissingFiles to optional? That way I can write config.AddJsonFile("file.json", optional: true)